### PR TITLE
✨ Add tuning variables for elasticity

### PR DESCRIPTION
**Pull request by AssassinTee**
Currently we are using nice fixed values for elasticity like 0.0 for players and 0.5 for flags. I am not a big fan of magic numbers, so I added tuning variables for them. I tested it with gametype `mod` and everything works fine, as long as you don't go over 1.0 for the Y-direction and 1.1 for the X-direction. Otherwise it's possible to clip through walls, but changing tuning values to unrealistic values is bound to break the game.

Why am I doing this? With the introduction of materials from other PRs like #2790 I simply want to have more playground with physics. This would enable bouncy blocks which could work really nice together with ice!

Also you might only want to bounce on the vertical edges (opposed to the ground you are standing on). That's why I introduced two variables for the player.

Edit: I also tested negative values. These make player stick to walls and a value of -1.0 is actually really funny, because you can stick to the ceiling for a bit. Everything beyond -1.0 is still broken.

screenshot_2022-03-27_12-45-53

The vertical shaft does not allow values over 1.1, because you can gain more speed with every bounce. On wider surfaces this is not as much of a problem and values up to 2.0 are possible. Only the cascading speed causes bugs

### CODE REVIEW

File: src/game/tuning.h [hunk 1]
```diff
@@ -23,6 +23,10 @@ MACRO_TUNING_PARAM(VelrampStart, velramp_start, 550)
 MACRO_TUNING_PARAM(VelrampRange, velramp_range, 2000)
 MACRO_TUNING_PARAM(VelrampCurvature, velramp_curvature, 1.4f)
 
+MACRO_TUNING_PARAM(PlayerElasticityX, player_elasticity_x, 0.0f)
+MACRO_TUNING_PARAM(PlayerElasticityY, player_elasticity_y, 0.0f)
+MACRO_TUNING_PARAM(FlagElasticity, flag_elasticity, 0.5f)
```
Comment (fokkonaut) [inline src/game/tuning.h:28]: Why would you not add one for flag x/y if you do so for the player :D?

Comment (AssassinTee) [reply src/game/tuning.h:28]: >  I simply want to have more playground with physics

😆 try them out, ground bouncyness is weird and wall bouncyness has much more potential IMO. The flag howerever can **only fall** (in defaul physics) and thus only has bouncyness in Y direction

**Comment by AssassinTee** [comment]
I added the advancement i made for ddnet, where this is already pushed, here, too.
